### PR TITLE
except tables not working with postgresql schema prefix

### PR DESCRIPTION
**Pull request by phoenixek12**
I define tables as strings containing only name of table and in postgres i get schema prefix in name. So simple array - not working for me. With this change except work on postges. Its a bit slower, but it keeps schema prefix and works as expected.

### CODE REVIEW

File: lib/database_cleaner/active_record/truncation.rb [hunk 1]
```diff
@@ -242,7 +242,14 @@ def clean
 
     def tables_to_truncate(connection)
       tables_in_db = cache_tables? ? connection.database_cleaner_table_cache : connection.tables
-      (@only || tables_in_db) - @tables_to_exclude - connection.database_cleaner_view_cache
+      to_reject = (@tables_to_exclude + connection.database_cleaner_view_cache)
+      (@only || tables_in_db).reject do |table|
+          if table.match(/([^.]+)$/)
+            to_reject.include?($1)
```
Comment (vrinek) [inline lib/database_cleaner/active_record/truncation.rb]: Please do not use `$1`. These types of variables are too cryptic for most developers and there are always ways around them. In this case, please assign the result from `table.match(/([^.]+)$/)` to a local variable and use that one.

Comment (phoenixek12) [inline lib/database_cleaner/active_record/truncation.rb]: Ok, I did it that way. Is it ok now?

**Comment by phoenixek12** [comment]
@vrinek Hi, what about this pull request? :(

**Comment by vrinek** [comment]
I'm sorry, I haven't found time to test it. I'll try to test it soon and merge it.

**Comment by etagwerker** [comment]
@phoenixek12 Could you add a scenario that shows the problem you're seeing?

**Comment by phoenixek12** [comment]
@etagwerker Is this good enough?

**Comment by etagwerker** [comment]
@phoenixek12 Yes, thanks!

**Pull request merged**